ssl/s3_lib.c

```diff
@@ -4723,12 +4736,26 @@ EVP_PKEY *ssl_generate_pkey_group(SSL *s, uint16_t id)
                  ERR_R_EVP_LIB);
         goto err;
     }
-    if (gtype != TLS_CURVE_CUSTOM
+# ifndef OPENSSL_NO_DH
+    if (gtype == TLS_GROUP_FFDHE
+            && (!(pkey = EVP_PKEY_new())
```

Comment:
This is a rather complicated expression.  Consider using two if statements (one for the gtype test, one for everything else).  At a minimum, the NULL test needs fixing and the nested indents are wrong.

Reply:
Changed to two if statements.